nfc/NfcCx/Cx/NfcCxSC.cpp

```diff
@@ -3669,3 +3801,133 @@ Routine Description:
 
     return tck;
 }
+
+BOOLEAN
+NfcCxSCInterfaceIsMultiProtocolTag(
+    _In_ PNFCCX_SC_INTERFACE ScInterface
+)
+/*++
+
+Routine Description:
+
+This routine checks wheather smart card tag is multiprotocol Tag
+
+--*/
+{
+    BOOLEAN fResult = FALSE;
+
+    TRACE_FUNCTION_ENTRY(LEVEL_VERBOSE);
+
+    NT_ASSERT(ScInterface->SmartCardConnected);
+
+    if (ScInterface->FdoContext->RFInterface->pLibNfcContext != NULL)
+    {
+        if (ScInterface->FdoContext->RFInterface->pLibNfcContext->bIsMultiProtocolTag == TRUE &&
+            ScInterface->FdoContext->RFInterface->pLibNfcContext->uNoRemoteDevices > 1)
+        {
+            fResult = TRUE;
+        }
+    }
+
+    TRACE_FUNCTION_EXIT(LEVEL_VERBOSE);
+
+    return fResult;
+}
+
+_Requires_lock_not_held_(ScInterface->SmartCardLock)
+NTSTATUS
+NfcCxSCInterfaceDeactivateMultiProtocolTag(
+    _In_ PNFCCX_SC_INTERFACE ScInterface
+)
+/*++
+
+Routine Description:
+
+This routine DeActivates the smart card
+
+Arguments:
+
+ScInterface - The SC Interface
+
+Return Value:
+
+NTSTATUS
+
+--*/
+{
+    NTSTATUS status = STATUS_SUCCESS;
+
+    TRACE_FUNCTION_ENTRY(LEVEL_VERBOSE);
+
+    TRACE_LINE(LEVEL_INFO, "Card DeactivateCommand Sent");
+
+    status = NfcCxRFInterfaceTargetDeactivate(ScInterface->FdoContext->RFInterface);
+
+    TRACE_FUNCTION_EXIT_NTSTATUS(LEVEL_VERBOSE, status);
+
+    return status;
+}
+
+_Requires_lock_not_held_(ScInterface->SmartCardLock)
+NTSTATUS
+NfcCxSCInterfaceActivateMultiProtocolTag(
+    _In_ PNFCCX_SC_INTERFACE ScInterface
+)
+/*++
+
+Routine Description:
+
+This routine Activates the smart card
+
+Arguments:
+
+ScInterface - The SC Interface
+
+Return Value:
+
+NTSTATUS
+
+--*/
+{
+    NTSTATUS status = STATUS_SUCCESS;
+
+    TRACE_FUNCTION_ENTRY(LEVEL_VERBOSE);
+
+    TRACE_LINE(LEVEL_INFO, "Card ActivateCommand Sent");
+
+    status = NfcCxRFInterfaceTargetActivate(ScInterface->FdoContext->RFInterface);
+
+    TRACE_FUNCTION_EXIT_NTSTATUS(LEVEL_VERBOSE, status);
+
+    return status;
+}
+
+DWORD
+NfcCxSCInterfaceGetIndexOfSelectedProtocol(
+    _Inout_ PNFCCX_RF_INTERFACE RFInterface,
+    _In_ phNfc_eRemDevType_t RemDevType
+)
```

Comment:
Method name NfcCxSCInterfaceGetIndexOfSelectedProtocol() might make false impression that it should return the value from RFInterface->pLibNfcContext->SelectedProtocolIndex. Considering RemDevType parameter, renaming it would improve readability. For example: 
NfcCxSCInterfaceGetIndexOfSelectedProtocol --> NfcCxSCInterfaceGetIndexOfProtocolType
NfcCxSCInterfaceLoadSelectedProtocol --> NfcCxSCInterfaceLoadNewSelectedProtocol